File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int carpet = brown + yellow;
        for(int i = 3; i <= carpet; i++) {
            int col = i;
            int row = carpet / i;
            if(row < 3) continue; //가로 3이상 이어야함
            if (row >= col && (row-2)*(col-2) == yellow) {
                answer[0] = row;
                answer[1] = col;
                break;
                
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    const total = brown + yellow;
    for (let w = total - 1; w >= 1; w--) {
        if (total % w !== 0) continue;
        const h = total / w;
        const y = (w - 2)*(h - 2);
        const b = total - y;
        if (brown === b && yellow === y) {
            answer[0] = w;
            answer[1] = h;
            break;
        }
    }
    return answer;
}